src/main/java/com/linecorp/armeria/server/docs/Specification.java

```diff
@@ -50,7 +50,7 @@ static Specification forServiceEntries(List<ServiceEntry> services) {
 
             final ThriftService thriftService = thriftServiceOptional.get();
             final Optional<String> debugPath =
-                    thriftService.serializationFormat() == SerializationFormat.THRIFT_TEXT ?
+                    thriftService.allowedSerializationFormats().contains(SerializationFormat.THRIFT_TEXT) ?
```

Comment:
Instead we could add the list of the allowed serialization formats and the path of the service (when exact path is available in the mapping) to the `specification.json`? The frontend could check if service has its path and supports TTEXT protocol before showing the form, which is slightly more work than before.